pkg/operator/ceph/object/spec.go

```diff
@@ -912,3 +922,93 @@ func renderProbe(cfg rgwProbeConfig) (string, error) {
+	dnsNames := c.store.Spec.Hosting.DNSNames
+	dnsNames = append(dnsNames, domainNameOfService(c.store))
+
+	// add custom endpoints from zone spec if exists
+	if c.store.Spec.Zone.Name != "" {
+		zone, err := c.context.RookClientset.CephV1().CephObjectZones(c.store.Namespace).Get(c.clusterInfo.Context, c.store.Spec.Zone.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		dnsNames = append(dnsNames, zone.Spec.CustomEndpoints...)
+	}
+
+	// validate dns names
+	var hostNames []string
+	for _, dnsName := range dnsNames {
+		hostName, err := GetHostnameFromEndpoint(dnsName)
+		if err != nil {
+			return "", err
+		}
+		hostNames = append(hostNames, hostName)
+	}
+
+	// remove duplicate host names
+	checkDuplicate := make(map[string]bool)
+	removeDuplicateHostNames := []string{}
+	for _, hostName := range hostNames {
+		if _, ok := checkDuplicate[hostName]; !ok {
+			checkDuplicate[hostName] = true
+			removeDuplicateHostNames = append(removeDuplicateHostNames, hostName)
+		}
+	}
+
+	return cephconfig.NewFlag("rgw dns name", strings.Join(removeDuplicateHostNames, ",")), nil
+}
+
+func GetHostnameFromEndpoint(endpoint string) (string, error) {
+	if len(endpoint) == 0 {
+		return "", fmt.Errorf("endpoint cannot be empty string")
+	}
+
+	// if endpoint doesn't end in '/', Ceph adds it
+	// Rook can do this also to get more accurate error results from this function
+	if !strings.HasSuffix(endpoint, "/") {
+		endpoint = endpoint + "/"
+	}
+
+	// url.Parse() requires a protocol to parse the host name correctly
+	if !strings.Contains(endpoint, "://") {
+		endpoint = "http://" + endpoint
+	}
+
+	// "net/url".Parse() assumes that a URL is a "path" with optional things surrounding it.
+	// For Ceph RGWs, we assume an endpoint is a "hostname" with optional things surrounding it.
+	// Because of this difference in fundamental assumption, Rook needs to adjust some endpoints
+	// used as input to url.Parse() to allow the function to extract a hostname reliably. Also,
+	// Rook needs to look at several parts of the url.Parse() output to identify more failure scenarios
+	parsedURL, err := url.Parse(endpoint)
+	if err != nil {
+		return "", err
+	}
+
+	logger.Debugf("parsedURL : %#v", parsedURL)
+
+	// error in this case: url.Parse("https://http://hostname") parses without error with `Host = "http:"`
+	// also catches issue where user adds colon but no port number after
+	if strings.HasSuffix(parsedURL.Host, ":") {
+		return "", fmt.Errorf("host %q parsed from endpoint %q has invalid colon (:) placement", parsedURL.Host, endpoint)
+	}
+
+	// as a convenience, allow users to give endpoints that have capitals
+	h := strings.ToLower(parsedURL.Hostname())
+
+	dnsErrs := validation.IsDNS1123Subdomain(h)
+	if len(dnsErrs) > 0 {
+		return "", fmt.Errorf("hostname %q parsed from endpoint %q is not a valid DNS-1123 subdomain: %v", h, endpoint, strings.Join(dnsErrs, ", "))
+	}
+
+	// pass original hostname back to caller
+	return parsedURL.Hostname(), nil
```

Comment:
This is returning the result that has not been lower-cased, but the convenience lower-casing line is still present. Is this intentional?

Reply:
Removed handling converting uppercase to lower case since ingress service follows RFC 1123